nexus/src/external_api/http_entrypoints.rs

```diff
@@ -7149,10 +7149,20 @@ impl NexusExternalApi for NexusExternalApiImpl {
             let opctx =
                 crate::context::op_context_for_external_api(&rqctx).await?;
             let user = nexus.silo_user_fetch_self(&opctx).await?;
-            let (_, silo) = nexus.current_silo_lookup(&opctx)?.fetch().await?;
+            let (authz_silo, silo) =
+                nexus.current_silo_lookup(&opctx)?.fetch().await?;
+
             Ok(HttpResponseOk(views::CurrentUser {
                 user: user.into(),
                 silo_name: silo.name().clone(),
+                fleet_viewer: opctx
+                    .authorize(authz::Action::Read, &authz::FLEET)
+                    .await
+                    .is_ok(),
+                silo_admin: opctx
+                    .authorize(authz::Action::Modify, &authz_silo)
+                    .await
+                    .is_ok(),
```

Comment:
What about a separate field (or array) for each action type? will clients ever need more information, like "what if this user has read but not modify on a silo?"

Reply:
Currently everyone has silo viewer on their silo. The distinction between silo collaborator and silo admin could be come relevant (or fleet viewer and fleet admin), but I can't think of a case in the client where we need to know that yet. I am inclined to not bother until we need it.